### PR TITLE
Add publisher for network

### PR DESCRIPTION
**Pull request by ar4s**

### CODE REVIEW

File: src/ralph/export_to_ng/publishers.py [hunk 1]
```diff
@@ -45,6 +46,7 @@ def wrapped_func(sender, instance=None, **kwargs):
                 # `_handle_post_save` set to False
                 getattr(instance, '_handle_post_save', True)
             ):
+                result = func(sender, instance, **kwargs)
```
Comment (ar4s) [inline src/ralph/export_to_ng/publishers.py]: TODO: delete this

File: src/ralph/export_to_ng/publishers.py [hunk 2]
```diff
@@ -195,3 +197,39 @@ def sync_virtual_server_to_ralph3(sender, instance=None, created=False, **kwargs
...
+            yield str(ipaddr.IPAddress(int_ip))
+        # end - 1 , end is reserved for broadcast address
+        for int_ip in xrange(end - 1, end - top - 1, -1):
+            yield str(ipaddr.IPAddress(int_ip))
+
+    def get_ids(manager):
+        return list(manager.all().values_list('id', flat=True))
+
+    return {
+        'address': net.address,
```
Comment (mkurek) [inline src/ralph/export_to_ng/publishers.py:279]: what about id?

File: src/ralph/export_to_ng/publishers.py [hunk 3]
```diff
@@ -195,3 +197,39 @@ def sync_virtual_server_to_ralph3(sender, instance=None, created=False, **kwargs
...
+        'remarks': net.remarks,
+        'vlan': net.vlan,
+        'dhcp_broadcast': net.dhcp_broadcast,
+        'gateway': net.gateway,
+        'reserved_ips': list(get_reserved_ips(net)) if net.min_ip and net.max_ip else [],  # noqa
+        'environment_id': net.environment_id,
+        'kind_id': net.kind_id,
+        'racks_ids': get_ids(net.racks),
```
Comment (mkurek) [inline src/ralph/export_to_ng/publishers.py]: rack is device in ralph2, but dedicated model in R3. Use sth similar to https://github.com/allegro/ralph/blob/develop/src/ralph/export_to_ng/resources.py#L508

File: src/ralph/export_to_ng/publishers.py [hunk 4]
```diff
@@ -195,3 +197,39 @@ def sync_virtual_server_to_ralph3(sender, instance=None, created=False, **kwargs
...
+        'dns_servers_ids': get_ids(net.custom_dns_servers),
```
Comment (mkurek) [inline src/ralph/export_to_ng/publishers.py]: maybe list of ip addresses will be enough here?

File: src/ralph/export_to_ng/publishers.py [hunk 5]
```diff
@@ -195,3 +201,81 @@ def sync_virtual_server_to_ralph3(sender, instance=None, created=False, **kwargs
...
+        template = instance.hosts_naming_template
+        counter_result = re.search('<([0-9]+),([0-9]+)>', template)
+        if not counter_result:
+            logger.info(
+                'Incorrect template for network environment with id {}. Return default values.'.format(instance.id)  # noqa
+            )
+            return {
+                'hostname_template_prefix': '',
+                'hostname_template_counter_length': 4,
+                'hostname_template_postfix': instance.domain or ''
```
Comment (mkurek) [inline src/ralph/export_to_ng/publishers.py]: maybe `'.{}'.format(instance.domain)` - in your case we'll have host `1234mydc.net`, assuming that domain name is `mydc.net`

File: src/ralph/export_to_ng/management/commands/sync_to_ralph3.py [hunk 6]
```diff
@@ -20,15 +20,21 @@
...
+)
 # register handlers
 from ralph.export_to_ng import publishers  # noqa
 
 logger = logging.getLogger(__name__)
 
 
 def generic_sync(model, **options):
-    for obj in model._default_manager.all():
+    for obj in model._default_manager.all()[:10]:
```
Comment (mkurek) [inline src/ralph/export_to_ng/management/commands/sync_to_ralph3.py]: rm limit

**Comment by mkurek** [comment]
:ok:

File: src/ralph/export_to_ng/publishers.py [hunk 7]
```diff
@@ -197,6 +203,93 @@ def sync_virtual_server_to_ralph3(sender, instance=None, created=False, **kwargs
...
+        'dhcp_broadcast': net.dhcp_broadcast,
+        'gateway': net.gateway,
+        'reserved_ips': list(get_reserved_ips(net)) if net.min_ip and net.max_ip else [],  # noqa
+        'environment_id': net.environment_id,
+        'kind_id': net.kind_id,
+        'racks_ids': list(get_racks_ids(net)) if net.racks.count() else [],
+        'dns_servers': list(
+            net.custom_dns_servers.all().values_list('ip_address', flat=True)  # noqa
+        ),
+
```
Comment (mkurek) [inline src/ralph/export_to_ng/publishers.py:292]: missing `}`

**Pull request merged**